src/dependency-check.js

```diff
@@ -9,25 +9,75 @@ const merge = require('merge-options')
 
 const defaultInput = [
   'package.json',
+  '.aegir.js',
   './test/**/*.js',
   './src/**/*.js',
   '!./test/fixtures/**/*.js'
 ]
 
+const commandNames = ['dependency-check', 'dep-check', 'dep']
+
+/**
+ * Returns true if the user invoked the command with non-flag or
+ * optional args
+ *
+ * @param {Array} args - process.argv or similar
+ * @returns {boolean} - true if the user passed files as positional arguments
+ */
+const hasPassedFileArgs = (args) => {
+  let foundCommand = false
+
+  for (let i = 0; i < args.length; i++) {
+    const arg = args[i]
+
+    if (arg === '--') {
+      // reached forward args
+      break
+    }
+
+    if (foundCommand && !arg.startsWith('-')) {
+      // was not a flag (e.g. -f) or option (e.g. --foo)
+      return true
+    }
+
+    if (commandNames.includes(arg)) {
+      foundCommand = true
+    }
+  }
+
+  return false
+}
+
 /**
  * Check dependencies
  *
  * @param {object} argv - Command line arguments passed to the process.
+ * @param {Array} processArgs - Unparsed command line arguments used to start the process
  * @param {ExecaOptions} execaOptions - execa options.
  * @returns {ExecaChildProcess} - Child process that does dependency check.
  */
-const check = (argv = { _: [] }, execaOptions) => {
-  const input = argv._.slice(1)
+const check = (argv = { _: [], input: [] }, processArgs = [], execaOptions) => {
   const forwardOptions = argv['--'] ? argv['--'] : []
-  const defaults = input.length ? input : defaultInput
+  let input = argv.input
+
+  if (argv.productionOnly) {
+    if (!hasPassedFileArgs(processArgs)) {
```

Comment:
yargs can't handle this ?

Reply:
No, from what I can see, there's no way to tell if a value came from a passed argument or if it came from the defaults. 😞

Reply:
the defaults are only used below so input should be empty right  ?

Reply:
`input` won't be empty - it's either the default value passed by yargs, or the input passed by the user

Reply:
ahhhh ok got it